### ISSUE TITLE
QGIS 3.26.1 on Python 3.11/Fedora 37 not working

### ISSUE THREAD

**Issue by nilsnolde** [NationalSecurityAgency/qgis-latlontools-plugin#71]
When activating the plugin in the plugin manager, QGIS crashes for me. I guess it has smth to do with python 3.11, it runs fine on another machine with python 3.10 and QGIS 3.28.x.

From a backtrace it seems to be QGIS/proj related, so nothing specific to this plugin probably:
```
{   "address": 139629838197668
                ,   "build_id": "8c5288cafaf6c45c5be8bf3a3ab3e4c760d4079d"
                ,   "build_id_offset": 517028
                ,   "function_name": "osgeo::proj::common::Measure::Measure(osgeo::proj::common::Measure const&) [clone .cold]"
                ,   "file_name": "/lib64/libproj.so.25"
                }
              , {   "address": 139629839076196
                ,   "build_id": "8c5288cafaf6c45c5be8bf3a3ab3e4c760d4079d"
                ,   "build_id_offset": 1395556
                ,   "function_name": "osgeo::proj::common::Angle::Angle(double)"
                ,   "file_name": "/lib64/libproj.so.25"
                }
              , {   "address": 139622758506710
                ,   "build_id_offset": 811222
                ,   "file_name": "/home/nilsnolde/.local/lib/python3.11/site-packages/pyproj.libs/libproj-eb58884b.so.25.9.1.0"
                }
```

**Comment by hamiltoncj** [comment]
The underlying Lat Lon Tools code would use the proj package from the QGIS API calls. It sounds like the error message is from pyproj. Perhaps it has not been really tested with Python 3.11. There is not really anything I can do to help with this. I doubt it is Lat Lon Tools that is causing this other than using pyproj from QGIS API calls. I work on Windows and don't have any Linux instances.

**Comment by nilsnolde** [comment]
Yeah think so too. I'll try the dev list, there should be a few Fedora users. Just wanted to leave that here, in case someone else has the same problems.

**Comment by alphafox02** [comment]
When trying to install/activate lat long tools from the built in plugin manager or the latest release downloaded from here on 22.04 Ubuntu with stock python 3.10 regardless if it's the latest LTS QGIS or the Current Release, the following happens in the terminal and qgis crashes (haven't did a back trace, but maybe it's similar to what's mentioned above).

terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)

**Comment by hamiltoncj** [comment]
The version of Python for Windows QGIS is 3.9.5. Using Python 3.11 could possibly be incompatible with QGIS. For the Fedora issue it appears to be a problem with the actually QGIS API calls that make used of PROJ. 

@alphafox02 For the Ubuntu issue I would recommend looking at Issue #66 and Issue #70, It appears that this is due to eventually having incompatible Linux libraries. What I usually find out is that it is not just Lat Lon Tools, but other plugins also have problems. When Linux users update to a current Linux and reinstall QGIS, the problems seem to go away. 

Unfortunately, I am not a Linux user so I cant really test this myself, but Lat Lon Tools is a very popular plugin and many of those are Linux users and they are not having problems. I wish I had a better explanation on why you are experiencing problems. My only insight into is is what others have said.

**Comment by nilsnolde** [comment]
Don't worry, that happens, and has nothing to do with your code, I kinda knew what I was getting into when choosing Fedora (also no other distro would work on my laptop :roll_eyes: ).

@alphafox02 What happens when you do `import pyproj` in the QGIS console? For me that crashes it. I'm on `pyproj 3.4.1`, which is [super new](https://github.com/pyproj4/pyproj/releases/tag/3.4.1), so that'd explain the crashing. I doubt though that 22.04 has the newest release, also tons of CI tests run on 22.04 and I'm sure one of those import pyproj somewhere (might also be a upstream proj issue). Did you install it from the official QGIS package repo?

I'll find the time soon to file a bug upstream. Probably they don't test on Fedora other than their mingw builds. So I might have to wait for the bug fix, latest once it hits other distros. But 22.04 should always work I believe.

@hamiltoncj feel free to close, I'll link the QGIS issue here soon.

**Comment by nilsnolde** [comment]
Thank god it became much easier to file QGIS bugs, so here it is: https://github.com/qgis/QGIS/issues/51268

**Comment by nilsnolde** [comment]
General side note: it might be a good idea to write some tests maybe? I was avoiding that with my QGIS plugins for a long time too, then I finally took some time to look into it. It's a bit more involved than usual testing, and not sure how well that translates to Windows, but I took down some notes and made it a tutorial: https://gis-ops.com/qgis-3-plugin-tutorial-geocoding-with-nominatim-part-4/. It was actually fun in the end:) Totally understand if you're not that interested or rather wait for the community to contribute.

**Comment by alphafox02** [comment]
> Don't worry, that happens, and has nothing to do with your code, I kinda knew what I was getting into when choosing Fedora (also no other distro would work on my laptop roll_eyes ).
> 
> @alphafox02 What happens when you do `import pyproj` in the QGIS console? For me that crashes it. I'm on `pyproj 3.4.1`, which is [super new](https://github.com/pyproj4/pyproj/releases/tag/3.4.1), so that'd explain the crashing. I doubt though that 22.04 has the newest release, also tons of CI tests run on 22.04 and I'm sure one of those import pyproj somewhere (might also be a upstream proj issue). Did you install it from the official QGIS package repo?
> 
> I'll find the time soon to file a bug upstream. Probably they don't test on Fedora other than their mingw builds. So I might have to wait for the bug fix, latest once it hits other distros. But 22.04 should always work I believe.
> 
> @hamiltoncj feel free to close, I'll link the QGIS issue here soon.

Still new to QGIS, so I'll have to go look at where the console is at. I do see that I have pyproj 3.4.0 installed. What I did just go do is install a fresh QGIS from the 22.04 stock repos (3.22.4), but yesterday I was installing the latest 3.22.x and 3.28 manually from the ubuntu QGIS repos.

What I just now figured out though is that I get the same issue I mentioned above, except for when I use the lat long tools 3.3.13. I went version by version back until one installed. Now I've not actually used it yet, but it installed and is viewable in the QGIS panel.

**Comment by alphafox02** [comment]
So I just want to thank you for both for getting me pointed in the right direction. Turns out I somehow had pyproj installed from pip3 and via apt, I think.. because I did a pip3 uninstall pyproj and then tried to install python3-pyproj and it told me that 3.3.0 was already manually installed. 

Now even the latest lat long plugin works for me on 22.04 with 3.22.4 QGIS and I suspect it'll work with the latest LTS etc.
